tests/python/tank_test/tank_test_base.py

```diff
@@ -279,6 +286,7 @@ def setUp(self, parameters=None):
         mockgun.Shotgun.set_schema_paths(mockgun_schema_path, mockgun_schema_entity_path)
 
         self.tank_temp = TANK_TEMP
+        self.addCleanup(shutil.rmtree, self.tank_temp)
```

Comment:
please add to tearDown instead.

It feels like we are trying to delete this `self.tank_temp` in many different places - here, as part of the backup logic, as part of the `tearDownModule` - is there any reason we need all of these places?